action/protocol/staking/vote_bucket.go

```diff
@@ -213,8 +218,13 @@ func bucketKey(index uint64) []byte {
 	return append(key, byteutil.Uint64ToBytesBigEndian(index)...)
 }
 
-func calculateVoteWeight(c genesis.VoteWeightCalConsts, v *VoteBucket, selfStake bool) *big.Int {
+// CalculateVoteWeight calculates the vote weight
+func CalculateVoteWeight(c genesis.VoteWeightCalConsts, v *VoteBucket, selfStake bool) *big.Int {
 	remainingTime := v.StakedDuration.Seconds()
+	if !v.isNative() {
```

Comment:
for NFT bucket, does it have `v.StakedDuation`?

Reply:
This time, the `StakedDuration` field is also set for the NFT bucket, based on the `StakedDurationBlockNumber* blockInterval`.